### ISSUE TITLE
Function with variadic generics Parameter Pack causes runtime crash when called from an initialised class instance.

### ISSUE THREAD

**Issue by efalloon** [apple/swift#74113]
### Description

When attempting to use a parameter pack in a function for an initialised `class`, the application crashes (runtime); this issue does not occur when the function is defined as `static`. Moreover, when the instance (as seen in the example below called 'Test') uses a `struct` constructor instead, also does not crash the application/ playground.

**Steps to Reproduce:**
1. Create a Swift class with a static shared instance.
2. Define a function in the class that uses a parameter pack.
3. Call this function using the shared instance with a tuple of mixed-type parameters.

_I have reduced the code example to the bare minimum for reproduction purposes and tested the code on other devices with similar (if not the same) result._

### Reproduction

❌ Using a `class` constructor fails:
```swift
class Test {
    func `try`<each T>(item: (repeat each T)) {
        repeat print(each item)
    }
}

//execute the function (`Test.try(...)` when static)
Test().try(item: (
    "Hey",
    7,
    true
))
```

✅ ...but the following cases will always succeed:

- Using a `class` constructor, but the function in question is `static`:
    ```swift
    class Test {
        static func `try`<each T>(item: (repeat each T)) {
            repeat print(each item)
        }
    }
    ```
- Using a value type (`struct`):
    ```swift
    struct Test { ... }
    ```

### Stack dump

```text
Stack dump:
0.	Program arguments: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-frontend -frontend -interpret kp_class.swift -Xllvm -aarch64-use-tbi -enable-objc-interop -stack-check -sdk /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk -color-diagnostics -new-driver-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift-driver -empty-abi-descriptor -resource-dir /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift -module-name kp_class -disable-clang-spi -target-sdk-version 14.5 -target-sdk-name macosx14.5 -external-plugin-path /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/bin/swift-plugin-server -external-plugin-path /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/local/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.5.sdk/usr/bin/swift-plugin-server -external-plugin-path /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server -external-plugin-path /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/local/lib/swift/host/plugins#/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/usr/bin/swift-plugin-server -plugin-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/host/plugins -plugin-path /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/local/lib/swift/host/plugins
1.	Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
2.	Compiling with the current language version
3.	While running user code "kp_class.swift"
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  swift-frontend           0x000000010a18bf3c llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  swift-frontend           0x000000010a18b0f8 llvm::sys::RunSignalHandlers() + 112
2  swift-frontend           0x000000010a18c544 SignalHandler(int) + 360
3  libsystem_platform.dylib 0x000000019a297584 _sigtramp + 56
4  libsystem_platform.dylib 0x000000010d6f4320 _sigtramp + 18446744071348538836
5  libsystem_platform.dylib 0x000000010d6f407c _sigtramp + 18446744071348538160
6  swift-frontend           0x0000000104bb3e04 llvm::orc::runAsMain(int (*)(int, char**), llvm::ArrayRef<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>, llvm::Optional<llvm::StringRef>) + 1344
7  swift-frontend           0x0000000104aea01c swift::SwiftJIT::runMain(llvm::ArrayRef<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>) + 168
8  swift-frontend           0x0000000104af819c swift::RunImmediately(swift::CompilerInstance&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char>>>> const&, swift::IRGenOptions const&, swift::SILOptions const&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>&&) + 1028
9  swift-frontend           0x0000000104a8fe90 processCommandLineAndRunImmediately(swift::CompilerInstance&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>&&, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::FrontendObserver*, int&) + 624
10 swift-frontend           0x0000000104a8a210 performCompileStepsPostSILGen(swift::CompilerInstance&, std::__1::unique_ptr<swift::SILModule, std::__1::default_delete<swift::SILModule>>, llvm::PointerUnion<swift::ModuleDecl*, swift::SourceFile*>, swift::PrimarySpecificPaths const&, int&, swift::FrontendObserver*) + 1760
11 swift-frontend           0x0000000104a89a18 swift::performCompileStepsPostSema(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 2100
12 swift-frontend           0x0000000104a8df0c performCompile(swift::CompilerInstance&, int&, swift::FrontendObserver*) + 3616
13 swift-frontend           0x0000000104a8b6d0 swift::performFrontend(llvm::ArrayRef<char const*>, char const*, void*, swift::FrontendObserver*) + 4968
14 swift-frontend           0x0000000104a1ae8c swift::mainEntry(int, char const**) + 2612
15 dyld                     0x0000000199ede0e0 start + 2360
```

### Expected behavior

The function `try(...)` should execute without crashing and print each element of the tuple when using a reference type constructor of `class`.

### Environment

```
swift-driver version: 1.90.11.1 Apple Swift version 5.10 (swiftlang-5.10.0.13 clang-1500.3.9.4)
Target: arm64-apple-macosx14.0
```

**Comment by AnthonyLatsis** [comment]
Runs fine with recent snapshots. Please confirm.

**Comment by efalloon** [comment]
It appears that the problem has been patched, but did not make it into the 5.10 release; please refer to https://github.com/apple/swift/commit/5f0357fdb0dfe667422ab4c5888b2ef6daba7eca.
